src/main/java/meteordevelopment/meteorclient/systems/modules/world/VeinMiner.java

```diff
@@ -139,7 +151,14 @@ private boolean isMiningBlock(BlockPos pos) {
     @EventHandler
     private void onStartBreakingBlock(StartBreakingBlockEvent event) {
         BlockState state = mc.world.getBlockState(event.blockPos);
-        if (state.getHardness(mc.world, event.blockPos) < 0 || blacklist.get().contains(state.getBlock())) return;
+
+        if (mode.get() == ListMode.Whitelist) {
+            if (!selectedBlocks.get().contains(state.getBlock()))
```

Comment:
the `state.getHardness(mc.world, event.blockPos) < 0` check should be done regardless if the mode is on whitelist or blacklist

Reply:
But why? If I whitelist a block that has a hardness less than zero, doesn't that mean I intend to break it? Of course, if I'm in creative mode or have the ability to do so.

Reply:
Done.